src/haddock/libs/libalign.py

```diff
@@ -596,113 +627,272 @@ def align_strct(reference, model, output_path, lovoalign_exec=None):
-
-        aln_fname = Path(output_path, f"blosum62_{ref_chain}.aln")
-        log.debug(f"Writing alignment to {aln_fname.name}")
-        with open(aln_fname, "w") as fh:
-            fh.write(str(top_aln))
-        aligned_ref_segment, aligned_model_segment = top_aln.aligned
-
-        # this should always be true
-        assert len(aligned_ref_segment) == len(aligned_model_segment)
-
-        identity = (
-            str(top_aln).count("|") / float(min(len(seq_ref), len(seq_model)))
-            ) * 100
-
-        if not any(e for e in top_aln.aligned):
+    aligner = Align.PairwiseAligner()
+    aligner.substitution_matrix = substitution_matrices.load("BLOSUM62")
+    alns = aligner.align(seq_ref, seq_model)
+    top_aln = alns[0]
+
+    aln_denom = min(len(seq_ref), len(seq_model))
+    identity = (str(top_aln).count("|") / float(aln_denom)) * 100
+    
+    aln_ref_seg, aln_mod_seg = top_aln.aligned
+    # this should always be true
+    assert len(aln_ref_seg) == len(aln_mod_seg)
+
+    return identity, top_aln, aln_ref_seg, aln_mod_seg
+
+
+class SeqAlign:
+    """SeqAlign class."""
+
+    def __init__(
+            self,
+            ):
+        """Initialize the class."""
+        self.align_dic = {}  # the alignment dictionary, important for CAPRI
+        self.model2ref_chain_dict = {}  # model to reference chain dictionary
+        self.ref2model_chain_dict = {}  # reference to model chain dictionary
+        self.seqdic_ref = None  # reference sequence dictionary
+        self.seqdic_model = None  # model sequence dictionary
+        self.seqs_ref = {}  # reference sequences
+        self.seqs_model = {}  # model sequences
+        self.identities = []  # list of identity values
+        self.aln_model_segs = []  # list of aligned model segments
+        self.aln_ref_segs = []  # list of aligned reference segments
+        self.top_alns = []  # list of alignment objects
+
+    def postprocess_alignment(
+            self,
+            ref_ch,
+            mod_ch,
+            align_id
+            ):
+        """
+        Postprocess the alignment.
+
+        Parameters
+        ----------
+        ref_ch : str
+            reference chain
+        
+        mod_ch : str
+            model chain
+        
+        align_id : int
+            alignment id (index of the alignment)
+        """
+        self.align_dic[ref_ch] = {}
+        if not any(e for e in self.top_alns[align_id].aligned):
             # No alignment!
             log.warning(
-                f"No alignment for chain {ref_chain} is it protein/dna-rna? "
+                f"No alignment for chain {ref_ch} is it protein/dna-rna? "
                 "Matching sequentially"
                 )
             if all(
-                    "X" in s for s in seq_ref) and all(
-                    "X" in s for s in seq_model):
+                    "X" in s for s in self.seqs_ref[ref_ch]) and all(
+                    "X" in s for s in self.seqs_model[mod_ch]):
```

Comment:
Any chance these "X" endup being lowercase? If so best to use `.upper()` in the conditional

Reply:
absolutely no chance :)